repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java

```diff
@@ -0,0 +1,313 @@
+
+        private NodeService nodeService;
+        private NodeRef rootNode;
+        @Spy ContentStore mockContentStore;
+        ContentService contentService;
+        ContentStore contentStore;
+
+        @Before
+        public void before()
+        {
+                nodeService = (NodeService) applicationContext.getBean("NodeService");
+                transactionService = (TransactionService) applicationContext.getBean("TransactionService");
+                this.authenticationComponent = (AuthenticationComponent) applicationContext.getBean("authenticationComponent");
+                this.contentService = (ContentService) this.applicationContext.getBean("contentService");
+                this.contentStore = (ContentStore) ReflectionTestUtils.getField(contentService, "store");
+
+                FileContentStore fileContentStore = new FileContentStore(applicationContext, this.contentStore.getRootLocation());
+                mockContentStore = spy(fileContentStore);
+
+                when(mockContentStore.getSupportedStorageClasses()).thenReturn(Set.of(DEFAULT_SC, "Azure", "S3"));
+                ReflectionTestUtils.setField(contentService, "store", mockContentStore);
+
+                this.authenticationComponent.setSystemUserAsCurrentUser();
+
+                StoreRef storeRef = new StoreRef(StoreRef.PROTOCOL_WORKSPACE, "testStoreRef");
+                if (!nodeService.exists(storeRef))
+                {
+                        storeRef = nodeService.createStore(storeRef.getProtocol(), storeRef.getIdentifier());
+                }
+                rootNode = nodeService.getRootNode(storeRef);
+        }
+
+        @After
+        public void afterTestMethod()
+        {
+                ReflectionTestUtils.setField(contentService, "store", contentStore);
+        }
+
+        @Test
+        public void testDefaultGetSupportedStorageClasses()
+        {
+                ReflectionTestUtils.setField(contentService, "store", contentStore);
+                assertTrue("Current supported storage classes: " + contentService.getSupportedStorageClasses(),
+                        contentService.getSupportedStorageClasses().contains(DEFAULT_VALUE));
+        }
+
+        @Test
+        public void testGetSupportedStorageClasses()
+        {
+                assertTrue("Currently supported storage classes:" + contentService.getSupportedStorageClasses()  , contentService.getSupportedStorageClasses().contains(DEFAULT_SC));
+        }
+
+        @Test
+        public void getDefaultStorageClassesTransition()
+        {
+                ReflectionTestUtils.setField(contentService, "store", contentStore);
+                assertTrue("Expected DEFAULT_SC ", contentService.getStorageClassesTransitions().isEmpty());
+        }
+
+        @Test
+        public void getStorageClassesTransitions()
+        {
+                Set<String> key1 = Set.of("Default");
+                Set<String> key2 = Set.of("Warm");
```

Comment:
I think you were referring to [WORM](https://aws.amazon.com/blogs/storage/tag/worm/)?

Reply:
Fixed